chall04/pcuadrad.py

```diff
@@ -0,0 +1,80 @@
+#!/usr/bin/env python3
+
+import sys
+
+def error_message(filename, message):
+    print (sys.argv[0] + ": " + filename + ": " + message)
+
+def place_books(books, shelfs):
+    rest = 0
+    last_book = 0
+    for i in range(0, len(shelfs)):
+        begin = shelfs[i]
+        begin += rest
+        rest = 0
+        if begin < 0 and i is len(shelfs) - 1:
+            return False
+        for j in range(last_book, len(books)):
+            begin -= books[j]
+            if begin < 0:
+                rest = begin
+                last_book = j + 1
+                break
+            if j is len(books) - 1:
+                print (i + 1)
+                return True
+    print (i + 1)
+    return True
+
+def parse_file(txt):
+    i = 0
+    books = []
+    shelfs_int = []
+    for line in txt:
+        if i is 0:
+            shelfs = line.split(' ')
```

Comment:
It is not specified that the output can have several spaces to separate shelves, but it might, you should use `.split()` without separators.
```suggestion
            shelfs = line.split()
```
As stated in the [method's documentation](https://docs.python.org/3/library/stdtypes.html#str.split):
> If sep is not specified or is None, a different splitting algorithm is applied: runs of consecutive whitespace are regarded as a single separator, and the result will contain no empty strings at the start or end if the string has leading or trailing whitespace. Consequently, splitting an empty string or a string consisting of just whitespace with a None separator returns [].